apps/query-ui/queryui/Sycamore_Query.py

```diff
@@ -32,28 +37,43 @@ def show_schema(_client: SycamoreQueryClient, index: str):
 
 
 @st.fragment
-def show_code(code: str):
+def show_code(client: SycamoreQueryClient, code: str):
+    st.session_state.code = code
+
     with st.expander("View code"):
         code = st_ace(
-            value=code,
-            key="python",
+            value=st.session_state.code,
+            key=f"python_{hash(st.session_state.query)}",
             language="python",
             min_lines=20,
         )
+
         execute_button = st.button("Execute Code")
         if execute_button:
             code_locals: dict = {}
             try:
                 with st.spinner("Executing code..."):
-                    exec(code, globals(), code_locals)
+                    global_context: dict[str, Any] = {"context": client.context}
+                    f = io.StringIO()
+                    with redirect_stdout(f):
+                        exec(code, global_context)
+                    result_str = f.getvalue()
+                    print("Printing out the results...")
+                    print(result_str)
+                    print("Done printing out the results...")
+                st.subheader("Result", divider="rainbow")
+                st.markdown(result_str, unsafe_allow_html=True)
+                # exec(code, global_context)
             except Exception as e:
                 st.exception(e)
             if code_locals and "result" in code_locals:
                 st.subheader("Result", divider="rainbow")
                 st.success(code_locals["result"])
-            if st.session_state.trace_dir:
-                st.subheader("Traces", divider="blue")
-                util.show_query_traces(st.session_state.trace_dir, st.session_state.query_id)
+            # Commenting this out for now -- we don't have the right query_id for this to work TODO
```

Comment:
This is only in the case where we did code generation, right?

Reply:
Correct. If I understand the flow here correctly, query_id wouldn't be generated in this case.